### PR TITLE
cxf-rt-rs-security-oauth2: fix 'Potential null dereference'

### PR DESCRIPTION
**Pull request by amarkevich**

### CODE REVIEW

File: rt/rs/security/oauth-parent/oauth2/src/main/java/org/apache/cxf/rs/security/oauth2/grants/code/JPACodeDataProvider.java [hunk 1]
```diff
@@ -143,13 +143,13 @@ public void setCodeLifetime(long codeLifetime) {
         if (c == null && resourceOwnerSubject == null) {
             return em.createQuery("SELECT c FROM ServerAuthorizationCodeGrant c",
                     ServerAuthorizationCodeGrant.class);
-        } else if (c == null) {
+        } else if (c == null && resourceOwnerSubject != null) {
```
Comment (coheigea) [inline rt/rs/security/oauth-parent/oauth2/src/main/java/org/apache/cxf/rs/security/oauth2/grants/code/JPACodeDataProvider.java]: I'm not sure if this change is needed?

File: rt/rs/security/oauth-parent/oauth2/src/main/java/org/apache/cxf/rs/security/oauth2/provider/JPAOAuthDataProvider.java [hunk 2]
```diff
@@ -324,56 +325,37 @@ public Void execute(EntityManager em) {
...
-                            + " WHERE s.login = :login AND c.clientId = :clientId", RefreshToken.class)
-                    .setParameter("login", resourceOwnerSubject.getLogin())
-                    .setParameter("clientId", c.getClientId());
+        return getQuery("RefreshToken", c, resourceOwnerSubject, entityManager, RefreshToken.class);
+    }
+
+    private <T> TypedQuery<T> getQuery(String table, Client c, UserSubject resourceOwnerSubject,
+            EntityManager entityManager, Class<T> resultClass) {
+        StringBuilder query = new StringBuilder("SELECT t FROM ").append(table).append(" t");
+        Map<String, Object> paramaterMap = new HashMap<>();
```
Comment (coheigea) [inline rt/rs/security/oauth-parent/oauth2/src/main/java/org/apache/cxf/rs/security/oauth2/provider/JPAOAuthDataProvider.java]: "paramaterMap" -> "parameterMap"

File: rt/rs/security/oauth-parent/oauth2/src/main/java/org/apache/cxf/rs/security/oauth2/provider/JPAOAuthDataProvider.java [hunk 3]
```diff
@@ -324,56 +325,37 @@ public Void execute(EntityManager em) {
...
-            return entityManager.createQuery(
-                    "SELECT t FROM RefreshToken t"
-                            + " JOIN t.client c"
-                            + " WHERE c.clientId = :clientId", RefreshToken.class)
-                    .setParameter("clientId", c.getClientId());
-        } else {
-            return entityManager.createQuery(
-                    "SELECT t FROM RefreshToken t"
-                            + " JOIN t.subject s"
-                            + " JOIN t.client c"
```
Comment (coheigea) [inline rt/rs/security/oauth-parent/oauth2/src/main/java/org/apache/cxf/rs/security/oauth2/provider/JPAOAuthDataProvider.java]: The joins aren't necessary any more?

File: rt/rs/security/oauth-parent/oauth2/src/main/java/org/apache/cxf/rs/security/oauth2/grants/code/JPACodeDataProvider.java (hunk 1 above)
Comment (amarkevich) [reply rt/rs/security/oauth-parent/oauth2/src/main/java/org/apache/cxf/rs/security/oauth2/grants/code/JPACodeDataProvider.java]: there are 4 null/non-null combinations possible; added explicit check for all cases

File: rt/rs/security/oauth-parent/oauth2/src/main/java/org/apache/cxf/rs/security/oauth2/provider/JPAOAuthDataProvider.java (hunk 3 above)
Comment (amarkevich) [reply rt/rs/security/oauth-parent/oauth2/src/main/java/org/apache/cxf/rs/security/oauth2/provider/JPAOAuthDataProvider.java]: joins replaced by t.client.clientId / t.subject.login

**Pull request merged**